### ISSUE TITLE
Helm cli cannot use process substitution for its config 

### ISSUE THREAD

**Issue by rlewkowicz** [helm/helm#9183]
For reasons, I need to be able to use process substitution for helm. Before you ask, my use case is not this simple, it's just to demonstrate the issue.

```
helm --kubeconfig=<(cat /Users/[YOUR-USER]/.kube/config) status flux
WARNING: Kubernetes configuration file is group-readable. This is insecure. Location: /dev/fd/11
Error: Kubernetes cluster unreachable: Get "http://localhost:8080/version?timeout=32s": dial tcp [::1]:8080: connect: connection refused
```

**Comment by bacongobbler** [comment]
The kubeconfig flag takes a file path, not a YAML string.

**Comment by rlewkowicz** [comment]
@bacongobbler in bash it's called process substitution. https://tldp.org/LDP/abs/html/process-sub.html
```
ls <(echo hello)
cat <(echo hello)
```

the tldr is it takes the output of a command as a file descriptor and uses it as an input for any command.

```
kubectl --kubeconfig=<(cat /Users/rlewkowicz/.kube/config) version
Client Version: version.Info{Major:"1", Minor:"20", GitVersion:"v1.20.1", GitCommit:"c4d752765b3bbac2237bf87cf0b1c2e307844666", GitTreeState:"clean", BuildDate:"2020-12-18T12:09:25Z", GoVersion:"go1.15.5", Compiler:"gc", Platform:"darwin/amd64"}
Server Version: version.Info{Major:"1", Minor:"19", GitVersion:"v1.19.4", GitCommit:"d360454c9bcd1634cf4cc52d1867af5491dc9c5f", GitTreeState:"clean", BuildDate:"2020-11-11T13:09:17Z", GoVersion:"go1.15.2", Compiler:"gc", Platform:"linux/amd64"}
```
My example is above is contrived just for demonstration. In reality, I'm wrapping the binary with python and using file descriptors so I don't have to manage persistent files. If my application crashes, the system will clean up after it's self etc.

```
helm --kubeconfig=<(cat /Users/rlewkowicz/.kube/config) env
WARNING: Kubernetes configuration file is group-readable. This is insecure. Location: /dev/fd/11
HELM_BIN="helm"
HELM_CACHE_HOME="/Users/rlewkowicz/Library/Caches/helm"
HELM_CONFIG_HOME="/Users/rlewkowicz/Library/Preferences/helm"
HELM_DATA_HOME="/Users/rlewkowicz/Library/helm"
HELM_DEBUG="false"
HELM_KUBEAPISERVER=""
HELM_KUBEASGROUPS=""
HELM_KUBEASUSER=""
HELM_KUBECONTEXT=""
HELM_KUBETOKEN=""
HELM_MAX_HISTORY="10"
HELM_NAMESPACE="default"
HELM_PLUGINS="/Users/rlewkowicz/Library/helm/plugins"
HELM_REGISTRY_CONFIG="/Users/rlewkowicz/Library/Preferences/helm/registry.json"
HELM_REPOSITORY_CACHE="/Users/rlewkowicz/Library/Caches/helm/repository"
HELM_REPOSITORY_CONFIG="/Users/rlewkowicz/Library/Preferences/helm/repositories.yaml"
KUBECONFIG="/dev/fd/11"
```

helm sees it. Something weird is going on under the hood. I can't dtruss this on my mac because I can't disable sip. I can play with this on linux tomorrow. I was just hoping some one is like "Yea, this spawns children like x so it can't y" or something. Process substitution is pretty standard bash fare.

**Comment by bacongobbler** [comment]
Okay. Thank you for the clarification. Never heard of process substitution before.

Looking at the example above, it does appear as though Helm supports process substitution as it recognizes the file descriptor and picks it up. So is the earlier premise ("Helm CLI cannot use process substitution") incorrect? If so we should close this for now until you can determine what's causing your issue.

In some cases it can help to use the `--debug` and `-v 10` flags to see what's happening under the hood.

**Comment by rlewkowicz** [comment]
@bacongobbler Thats why I suggested trying a network call

```
helm --kubeconfig=/Users/[YOUR-USER]/.kube/config status [anything]
```

I did flux, but pick anything. It sees the file descriptor, but then when it goes to use it, it's gone. Then it randomly tries to connect to localhost:

```
Error: Kubernetes cluster unreachable: Get "http://localhost:8080/version?timeout=32s": dial tcp [::1]:8080: connect: connection refused
```

Please actually try this if you would. There's something severely wrong with the way helm manages its processes in comparison to essentially any other binary I've ever used.

**Comment by ryancastle** [comment]
@rlewkowicz Have you tried on more than one OS? 

This works on one Linux OS for me (Ubuntu/Bash 5.0), but not the one I actually have to use (CentOS/Bash 4.2). On CentOS, Helm just silently ignores the pipe altogether.

```
helm template -f <(sops -d values.yaml) ...
```
